### PR TITLE
First commit for console-gitops-plugin

### PR DESCRIPTION
**Pull request by ciiay**
For GITOPS-2059.

### CODE REVIEW

**Comment by keithchong** [comment]
Hi @ciiay , don't include the node_modules and the dist folders in the PR.

File: src/components/GitOpsListPage.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+import Helmet from 'react-helmet';
+import { useTranslation } from 'react-i18next';
+import { useK8sWatchResource, ListPageHeader, ListPageBody } from '@openshift-console/dynamic-plugin-sdk';
+
+import { fetchAllAppGroups, getManifestURLs } from './utils/gitops-utils';
+// import { fetchAppGroups, /* fetchAllAppGroups, getManifestURLs  getPipelinesBaseURI */ } from './utils/gitops-utils';
+import { GitOpsAppGroupData } from './utils//gitops-types';
+import GitOpsList from './list/GitOpsList';
+
+// import { PageHeading /*, LoadingBox */ } from '@console/internal/components/utils';
```
Comment (keithchong) [inline src/components/GitOpsListPage.tsx:12]: If leaving in and commenting out the code that we need to migrate or port helps with the port work, then I'm ok with it.  Just a not to make sure we are working with the latest and greatest from the static plugin and not from these commented out code.

File: console-extensions.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    "flags": {
+      "required": ["OPENSHIFT"]
+    }
+  },
+  {
+    "type": "console.page/route",
+    "properties": {
+      "component": { "$codeRef": "environments" },
+      "exact": false,
+      "path": "/envdynamic"
```
Comment (keithchong) [inline console-extensions.json:32]: Make a note we will change the path (and the href above) and the name later.

File: package.json [hunk 3]
```diff
@@ -0,0 +1,69 @@
+{
+  "name": "console-gitops-plugin",
+  "description": "OpenShift Console plugin for GitOps",
+  "version": "0.0.1",
+  "private": true,
+  "scripts": {
+    "build-ORIG": "yarn clean && NODE_ENV=production yarn ts-node ./node_modules/.bin/webpack --mode=production",
+    "build-dev-ORIG": "yarn clean && yarn ts-node ./node_modules/.bin/webpack --mode=development",
+    "http-server-ORIG": "./http-server.sh ./dist",
```
Comment (keithchong) [inline package.json:9]: What's the reason for changing these to -ORIG ?

File: package.json [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    "typescript": "^4.6.2",
+    "webpack": "^5.68.0",
+    "webpack-cli": "^4.9.2",
+    "webpack-dev-server": "^4.9.3"
+  },
+  "consolePlugin": {
+    "name": "console-gitops-plugin",
+    "version": "0.0.1",
+    "displayName": "GitOps Plugin",
+    "description": "OpenShift Console plugin for Gitops.",
```
Comment (keithchong) [inline package.json:61]: GitOps

**Comment by keithchong** [comment]
Hi @ciiay , I approve pending comments.

**Review by keithchong — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (ciiay) [reply package.json:9]: Will delete those.

**Pull request merged**